api/src/main/java/net/md_5/bungee/api/event/AsyncEvent.java

```diff
@@ -43,30 +42,44 @@ public void postCall()
     /**
      * Register an intent that this plugin will continue to perform work on a
      * background task, and wishes to let the event proceed once the registered
-     * background task has completed.
+     * background task has completed. Multiple intents can be registered by a
+     * plugin, but the plugin must complete the same amount of intents for the
+     * event to proceed.
      *
      * @param plugin the plugin registering this intent
      */
     public void registerIntent(Plugin plugin)
     {
         Preconditions.checkState( !fired.get(), "Event %s has already been fired", this );
-        Preconditions.checkState( !intents.contains( plugin ), "Plugin %s already registered intent for event %s", plugin, this );
 
-        intents.add( plugin );
+        AtomicInteger intentCount = intents.get( plugin );
```

Comment:
`intents.computeIfAbsent( plugin , (k) -> new AtomicInteger(0)).incrementAndGet();` ?

Reply:
Ok, your right

Reply:
That doesn't create a new atomicinteger every time

Reply:
Oh I meant that about putIfAbsent, not computeIfAbsent.